File: competitive_programming/codeforces/06_javascript/271a_beautiful_year.js
function beautifulYear(year) {
  let current = year;
  while (true) {
    current += 1;
    if (new Set(String(current)).size === 4) return current;
  }
}

if (require.main === module) {
  if (beautifulYear(1987) !== 2013 || beautifulYear(2013) !== 2014) throw new Error("beautiful year failed");
  console.log("271A beautiful year ok");
}

File: competitive_programming/codeforces/09_rust/271a_beautiful_year.rs
use std::collections::HashSet;

fn beautiful_year(year: i32) -> i32 {
    let mut current = year;
    loop {
        current += 1;
        let digits: HashSet<char> = current.to_string().chars().collect();
        if digits.len() == 4 {
            return current;
        }
    }
}

fn main() {
    assert_eq!(beautiful_year(1987), 2013);
    assert_eq!(beautiful_year(2013), 2014);
    println!("271A beautiful year ok");
}
